simplecpp.cpp

```diff
@@ -58,6 +58,15 @@ static bool isOct(const std::string &s)
     return s.size()>1 && (s[0]=='0') && (s[1] >= '0') && (s[1] < '8');
 }
 
+static bool isStringConstant(const std::string &s)
+{
+    return s.size() > 1 && (s[0]=='\"') && (*s.rbegin()=='\"');
+}
+
+static bool isCharConstant(const std::string &s)
+{
+    return s.size() == 3 && (s[0]=='\'') && (s[2]=='\'');
```

Comment:
imho this should also be classified as a char constant: `'\t'`.
a multi char constant is not handled: `'ab'`. is that intentional?

Reply:
Ah, good point (and now the comment about '\'' in the original thread makes way more sense).

Reply:
multichar constants don't seem to work with operator "" though, so that might be accidentally intentional. Will investigate further.

Reply:
So the pragmatic approach seems to be to relax the restriction to `s.size() > 2`. That covers `‘a’`, `‘1234’`, `‘\t’`, `‘\000’` etc. but doesn't check the validity, which seem in line with the regular output. Or even > 1 analog to strings since there’s the weird case of `’’`.

Multi-character plus user-defined operator "" don't seem to actually _compile_ but clang/gcc will happily preprocess it with `-E` ([godbolt](https://godbolt.org/z/GvoExP5Pj)).

Reply:
I didn't know about `''` , yes that is weird :-)

So yes I think the pragmatic approach sounds good.